src/iOS.Extension/LockPasswordViewController.cs

```diff
@@ -19,12 +19,20 @@ public LockPasswordViewController(IntPtr handle)
         public override UIBarButtonItem BaseSubmitButton => SubmitButton;
         public override Action Success => () => LoadingController.DismissLockAndContinue();
         public override Action Cancel => () => LoadingController.CompleteRequest(null, null);
+        public Action LaunchHomePage;
 
         public override void ViewDidLoad()
         {
             base.ViewDidLoad();
             CancelButton.TintColor = ThemeHelpers.NavBarTextColor;
             SubmitButton.TintColor = ThemeHelpers.NavBarTextColor;
+
+            if (!HasLoginOrUnlockMethod)
+            {
+                // user doesn't have a login method
+                // needs to go to homepage and login again
+                LaunchHomePage?.Invoke();
+            }
```

Comment:
🤔 Can we change the inheritance to use the same base lock VC so this doesn't have to be repeated here? If not also transform the `LaunchHomePage` into a property. I have to do a refactor for all extensions so I can tackle changing the base there as well.